pkg/providers/pricing/suite_test.go

```diff
@@ -252,3 +258,34 @@ var _ = Describe("Pricing", func() {
 			To(ContainElements("Linux/UNIX", "Linux/UNIX (Amazon VPC)"))
 	})
 })
+
+func getPricingEstimateMetricValue(instanceType string, capacityType string, zone string) float64 {
+	gathering, err := crmetrics.Registry.Gather()
+	Expect(err).To(BeNil())
+	var value *float64
+	for _, mf := range gathering {
+		if mf.GetName() == "karpenter_pricing_instance_estimate" {
```

Comment:
Can we short circuit this logic so the remainder of the code get less nested?
```suggestion
		if mf.GetName() != "karpenter_pricing_instance_estimate" {
		    continue
		}
```

Reply:
Fixed this.